### PR TITLE
Policy Changes (Operator -> Manifest) 

### PR DESCRIPTION
**Pull request by pasanw**
Use this PR to visualize the changes made by the `allow-tigera-reconciliation` branch to rendered policy.

The LHS of the diff view represents what policy looks like once rendered by the operator. The RHS represents the current state of policy, i.e. policy prior to operator.

The LHS is simply the test expectations that the render components use to validate all possible render scenarios. The RHS are the current policy manifests converted to JSON.

### CODE REVIEW

File: pkg/render/testutils/expected_policies/compliance_managed.json [hunk 1]
```diff
@@ -17,10 +17,13 @@
         "action": "Allow",
         "protocol": "TCP",
         "destination": {
-          "services": {
```
Comment (caseydavenport) [inline pkg/render/testutils/expected_policies/compliance_managed.json:20]: Why this change?

File: pkg/render/testutils/expected_policies/dpi_managed_ocp.json [hunk 2]
```diff
@@ -33,16 +33,6 @@
           }
         }
       },
-      {
```
Comment (caseydavenport) [inline pkg/render/testutils/expected_policies/dpi_managed_ocp.json:36]: Do we not need access to DNS any more? Is this covered by another rule (sorry if this is obvious and I'm just missing or forgetting)

Comment (pasanw) [reply pkg/render/testutils/expected_policies/dpi_managed_ocp.json:36]: The operator implementation is adding (what I assume to be) a missing DNS TCP allow in the OCP manifests for DPI

File: pkg/render/testutils/expected_policies/compliance_managed.json (hunk 1 above)
Comment (pasanw) [reply pkg/render/testutils/expected_policies/compliance_managed.json:20]: The operator implementation is correcting (what I assume to be) an inconsistency between the use of service selectors in the managed and unmanaged variant of this policy
